docs/0-Preparing/starting_side_projects.md

```diff
@@ -0,0 +1,93 @@
+
+> You might have heard that the tech industry changes quickly, trends come and go and technologies become obselete quickly. In addition, many companies have proprietary tools internal to the company. That's why it doesn't really matter *what* you learn, as long as you learn *something*. That signals to employers that you can learn whatever is necessary on the job.
+
+### Becoming and staying motivated
+
+It's ideal that you only do side projects if you're enjoying it. That way, it's sustainable, and just a generally healthy thing to do.
+
+But realistically, expect that it can take a few tries to enjoy something. Not all aspects of programming appeal to everyone. It's possible that the stuff you learned in class so far is pretty dry. Some people will enjoy UI/visual/graphics programming. Some people like math and algorithms. Some people like systems. Some people like building products. Some people like to do social good via tech. If you've committed to programming in some way anyway (e.g. enrolled in a CS degree), might as well do some exploration to see what you really like.
+
+### Project ideas
+See this [great GitHub repo](https://github.com/karan/Projects) with tons of project ideas.
+
+**Online courses**
+
+This is often the easiest way to start. You don't need to come up with an idea or even know anything, you can just go around shopping for something that looks interesting. We know, we know, you're taking enough classes at school already. But these courses are often more practical, fun, and have more selection.
+
+They can also leave you with something to show in the end. They allow you to build something super fancy complicated. You can create something through a course (like this [Udacity web development course](https://www.udacity.com/course/web-development--cs253)) or make a game like hangman or build a calculator or whatever else you can think of. You'll be surprised how complicated "simple projects" can be and how much you can learn. That's what employers want to see.
+
+Places to shop for courses: Udacity, Coursera, EdX.
+
+**Reimplementing stuff**
+
+A great way to learn how something is built is to build it. Make a simple Twitter clone, a BitTorrent, the `ls` command-line function, a compiler for a small subset of C, etc.
+
+**Solve a problem you or something you know has**
+
+Great ideas come when you notice a problem someone has, that could be solved with software. I don't expect that telling you something vague like this will generate an idea *right now*, but if you keep it in the back of your mind, an opportunity might come by!
+
+**Make a game**
+
+Games tend to especially satisfying to build because you get to play with it as you go. There are very few constraints, it's very free-form. They also tend to cover a wide range of programming concepts (AI algorithms, OOP design patterns, UI, etc).
```

Comment:
-> Along the way, you'll get a chance to learn from a wide range of programming concepts (AI algorithms, OOP design patterns, UI, etc).